src/main/resources/com/google/api/codegen/go/smoke.snip

```diff
@@ -0,0 +1,77 @@
+@extends "go/header.snip"
+@extends "go/initcode.snip"
+
+@snippet generate(view)
+  {@headerComment(view.fileHeader)}
+
+  package {@view.fileHeader.localPackageName}
+
+  import (
+    {@renderImportSection(view.fileHeader.importSection)}
+  )
+
+  import (
+    "cloud.google.com/go/internal/testutil"
+    "golang.org/x/net/context"
+    "google.golang.org/api/iterator"
+    "testing"
+  )
+
+  var _ = iterator.Done
+
+  func {@view.name}(t *testing.T) {
+    if testing.Short() {
+      t.Skip("skipping smoke test in short mode")
+    }
+
+    projectId := testutil.ProjID()
```

Comment:
For context: The `testutil` package is already used by handwritten clients. We're just reusing it.

> 2. We should allow the user to pass in the ID from command line

It is configurable with `GCLOUD_TESTS_GOLANG_PROJECT_ID` environment variable. So you can run
```shell
GCLOUD_TESTS_GOLANG_PROJECT_ID=my-project go test cloud.google.com/go/...
```
and test all handwritten and gapic clients with the same project.

> 1. projectId is not always needed for every test

It's OK. If you don't set the env var, `testutil.ProjID` returns empty string. If the test doesn't need it, it doesn't matter. It won't throw exception and fail the test.